tools/rtpsrelay/ParticipantListener.cpp

```diff
@@ -51,10 +53,11 @@ void ParticipantListener::on_data_available(DDS::DataReader_ptr reader)
         const auto p = guids_.insert(repoid);
         if (p.second) {
           if (config_.log_discovery()) {
-            ACE_DEBUG((LM_INFO, ACE_TEXT("(%P|%t) INFO: ParticipantListener::on_data_available add local participant %C %C\n"), guid_to_string(repoid).c_str(), OpenDDS::DCPS::to_json(data).c_str()));
+            const auto d = now - guid_addr_set_.get_first_spdp(repoid);
+            ACE_DEBUG((LM_INFO, ACE_TEXT("(%P|%t) INFO: ParticipantListener::on_data_available add local participant %C %C discovery time %d.%d\n"), guid_to_string(repoid).c_str(), OpenDDS::DCPS::to_json(data).c_str(), d.value().sec(), d.value().usec()));
```

Comment:
Check log formatting characters for these types.  `sec()` is `time_t` and `usec()` is `suseconds_t`

Reply:
Looks like it's `%:`?

Reply:
I don't see a formated for suseconds_t.

Reply:
Convert the argument to a type (`int`) that we know will match what the varargs macro is looking for based on the format (`%d`).